File: backend-csharp/Controllers/UserController.cs
using CollaborativeDashboard.Api.Models;
using CollaborativeDashboard.Api.Services;
using Microsoft.AspNetCore.Mvc;

namespace CollaborativeDashboard.Api.Controllers;

[ApiController]
[Route("api/users")]
public sealed class UsersController(UsersService service) : ControllerBase
{
    [HttpGet("by-userId/{userId}")]
    public Task<ApiResponse<UserSummary>> GetUserById(string userId, CancellationToken cancellationToken) => service.GetUserByIdAsync(userId, cancellationToken);

    [HttpGet("by-username/{username}")]
    public Task<ApiResponse<UserSummary>> GetUserByUsername(string username, CancellationToken cancellationToken) => service.GetUserByUsernameAsync(username, cancellationToken);

    [HttpGet("by-email/{email}")]
    public Task<ApiResponse<UserSummary>> GetUserByEmail(string email, CancellationToken cancellationToken) => service.GetUserByEmailAsync(email, cancellationToken);

    [HttpGet]
    public Task<ApiResponse<UserSummary>> GetAllUsers(CancellationToken cancellationToken) => service.GetAllUsersAsync(cancellationToken);

    [HttpPost]
    public Task<ApiResponse<UserSummary>> CreateUser(UserSummary userSummary, CancellationToken cancellationToken) => service.CreateUserAsync(userSummary, cancellationToken);

    [HttpDelete("delete/{userId}")]
    public Task<ApiResponse<UserSummary>> DeleteUserById(string userId, CancellationToken cancellationToken) => service.DeleteUserByIdAsync(userId, cancellationToken);
    
    [HttpPut("update/{userId}")]
    public Task<ApiResponse<UserSummary>> UpdateUserByUserId([FromBody] UserSummary userSummary, CancellationToken cancellationToken) => service.UpdateUserByUserId(userSummary, cancellationToken);
}

File: backend-java/src/main/java/com/collaborative/dashboard/controller/UserController.java
package com.collaborative.dashboard.controller;

import com.collaborative.dashboard.model.ApiResponse;
import com.collaborative.dashboard.model.UserSummary;
import com.collaborative.dashboard.service.UserService;
import java.util.List;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/users")
public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/by-userId/{userId}")
    public ApiResponse<List<UserSummary>> getUserById(@PathVariable String userId) {
        return userService.getUserById(userId);
    }

    @GetMapping("/by-username/{username}")
    public ApiResponse<List<UserSummary>> getUserByUsername(@PathVariable String username) {
        return userService.getUserByUsername(username);
    }

    @GetMapping("/by-email/{email}")
    public ApiResponse<List<UserSummary>> getUserByEmail(@PathVariable String email) {
        return userService.getUserByEmail(email);
    }

    @GetMapping
    public ApiResponse<List<UserSummary>> getAllUsers() {
        return userService.getAllUsers();
    }

    @PostMapping
    public ApiResponse<List<UserSummary>> createUser(@RequestBody UserSummary userSummary) {
        return userService.createUser(userSummary);
    }

    @DeleteMapping("/delete/{userId}")
    public ApiResponse<List<UserSummary>> deleteUserById(@PathVariable String userId) {
        return userService.deleteUserById(userId);
    }

    @PutMapping("/update/{userId}")
    public ApiResponse<List<UserSummary>> updateUserById(
            @PathVariable String userId, @RequestBody UserSummary userSummary) {
        return userService.updateUserById(userId, userSummary);
    }
}